### PR TITLE
Improve file fallback routing behavior when using ConsumesAttribute

### LINKED ISSUE #41060: MapFallback catch all requests to API controller when ConsumeAttribute added
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hello.
In .net 6 behavior of MapFallback broke when ConsumesAttribute was added. MapFallback catch all requests to API controller.
In .net 6 index.html file always returns on requests to API controller but if I remove endpoints.MapFallbackToFile or delete ConsumesAttribute filter then everything works as I expected.
**In .net 5 the example works correctly.**

ConsumesAttribute was added for the correct generation of swagger.json by Swashbuckle in order not to add it to every action or controller.

### Expected Behavior

Requests to controllers must not be catched by MapFallback.

### Steps To Reproduce

```c#
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllers(options => options.Filters.Add(new ConsumesAttribute("application/json")));
        }
        public void Configure(IApplicationBuilder app)
        {
            app.UseRouting();
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllers();
                endpoints.MapFallbackToFile("index.html");
            });
        }
    }
```

### Exceptions (if any)

_No response_

### .NET Version

6.0.200

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by MackinnonBuck**
# Improve file fallback routing behavior when using `ConsumesAttribute`

Prevents the configured fallback file from matching routes that were discarded due to a mismatch between the request body content type and the content type specified via a `ConsumesAttribute` on a controller action.

## Description

This fix works by only allowing the fallback file to be returned when making a GET request. As such, the fix specifically improves the behavior for non-GET controller actions.

Fixes #41060

## Customer Impact

Some customers have had their applications affected by this issue when migrating from .NET 5 to .NET 6. This fix reduces the magnitude of the breaking change in routing logic introduced in .NET 6, so customers are less likely to have their apps be negatively affected.

## Regression?

- [X] Yes
- [ ] No

Regressed from .NET 5.

## Risk

- [ ] High
- [ ] Medium
- [X] Low

This is a narrowly scoped change, and it's unlikely customers are relying on file fallback routing matching non-GET requests.

## Verification

- [X] Manual (required)
- [X] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

File: src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -13,6 +13,9 @@ namespace Microsoft.AspNetCore.Builder;
 /// </summary>
 public static class StaticFilesEndpointRouteBuilderExtensions
 {
+    // See: https://github.com/dotnet/aspnetcore/issues/41060
+    private static readonly string[] _supportedHttpMethods = new[] { HttpMethods.Get };
```
Comment (Tratcher) [inline src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs:17]: And HEAD?

File: src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -13,6 +13,9 @@ namespace Microsoft.AspNetCore.Builder;
 /// </summary>
 public static class StaticFilesEndpointRouteBuilderExtensions
 {
+    // See: https://github.com/dotnet/aspnetcore/issues/41060
```
Comment (Tratcher) [inline src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs:16]: Inline the relevant explanation.

**Review by javiercn — APPROVED** [review]
Looks great @MackinnonBuck!

**Comment by MackinnonBuck** [comment]
@Pilchie Would you please take a look? Thanks!

**Comment by Pilchie** [comment]
Approved for .NET 7 RC2.

File: src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs [hunk 3]
```diff
@@ -42,17 +48,12 @@ public static class StaticFilesEndpointRouteBuilderExtensions
...
 
-        if (filePath == null)
-        {
-            throw new ArgumentNullException(nameof(filePath));
-        }
-
-        return endpoints.MapFallback(CreateRequestDelegate(endpoints, filePath));
+        return endpoints
+            .MapFallback(CreateRequestDelegate(endpoints, filePath))
+            .WithMetadata(new HttpMethodMetadata(_supportedHttpMethods));
```
Comment (BrennanConroy) [inline src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs:56]: Do we know that this doesn't affect CORS?
See https://github.com/dotnet/aspnetcore/issues/42994 for an issue with CORS and HttpMethodMetadata

**Pull request merged**